src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -373,6 +401,94 @@ private void runFafReplayFile(Path path) throws IOException, CompressorException
     gameService.runWithReplay(tempSupComReplayFile, replayId, gameType, version, modVersions, simMods, mapName);
   }
 
+  private void runRehostOnlineReplay(int replayId) {
+    downloadReplay(replayId)
+        .thenAccept((path) -> {
+          try {
+            runRehostFile(path);
+          } catch (IOException | CompressorException e) {
+            throw new RuntimeException(e);
+          }
+        })
+        .exceptionally(throwable -> {
+          if (throwable.getCause() instanceof FileNotFoundException) {
```

Comment:
Make sure to unwrap the exception using something like ConcurrentUtil.unWrapException